lib/child-writer/index.js

```diff
@@ -113,25 +113,36 @@ exports.write = (image, drive, options) => {
     emitter.emit('error', error);
   };
 
-  ipc.server.on('error', emitError);
-  ipc.server.on('message', (data) => {
-    let message = null;
-
+  /**
+   * @summary Bridge robot message to the child writer caller
+   * @function
+   * @private
+   *
+   * @param {String} message - robot message
+   *
+   * @example
+   * bridgeRobotMessage(robot.buildErrorMessage(new Error('foo')));
+   */
+  const bridgeRobotMessage = (message) => {
     try {
-      message = robot.parseMessage(data);
-    } catch (error) {
-      return emitError(error);
-    }
+      const parsedMessage = robot.parseMessage(message);
+
+      // The error object is decomposed by the CLI for serialisation
+      // purposes. We compose it back to an `Error` here in order
+      // to provide better encapsulation.
+      if (robot.getCommand(parsedMessage) === 'error') {
+        emitError(robot.recomposeErrorMessage(parsedMessage));
+      } else {
+        emitter.emit(robot.getCommand(parsedMessage), robot.getData(parsedMessage));
```

Comment:
> Could we have a single function that returned both the command and data (e.g. as a tuple) at the same time?

Tuples are not very used in the JS world. We could return an object, but that'd be basically returning `parsedMessage` again :)

> Or maybe parsedMessage itself should be an Object with accessor functions (this is where my JS knowledge runs out) so you could do parsedMessage.getCommand() and parsedMessage.getData() ?

That would mean

```js
emitter.emit(robot.getCommand(parsedMessage), robot.getData(parsedMessage));
```

gets changed to:

```js
emitter.emit(parsedMessage.getCommand(), parsedMessage.getData());
```

Which is basically the same, plus I think we should go functional when possible.

> Alternatively, how likely is it that robot.getCommand(parsedMessage) will ever do anything other than returning parsedMessage.command ?

Probably nothing else. The main benefit of function getters is that JavaScript is a weakly typed language. We might accidentally do `message.comand`, which will silently be evaluated as `undefined`, and will not match any robot message handler, causing the data on that message to be ignored, etc. Having function getters protect us from that kind of things.